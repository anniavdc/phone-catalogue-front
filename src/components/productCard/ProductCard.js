import React from 'react';

export default function ProductCard(props) {
    const {
        title,
        info,
        imageUrl,
        content
    } = props;

    return (
        <div className="product-card">
            <p className="card-info">{info}</p>
            <div className="card-image" style={{ backgroundImage: `url(${imageUrl})` }}>
            </div>
            <div>
                <h3 className="card-title">{title}</h3>
                <p className="card-content">{content}</p>
            </div>
        </div>
    )
}
