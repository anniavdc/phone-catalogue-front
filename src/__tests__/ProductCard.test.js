import React from 'react';
import { render } from '@testing-library/react';

// files
import ProductCard from './../components/productCard/ProductCard';

const initialProps = {
    title: "Galaxy 7",
    info: 59,
    imageUrl: "https://image.png",
    content: "Samsung",
    onCardClick: jest.fn()
}

describe('ProductCard', () => {
    test('render', () => {
        const { container } = render(
            <ProductCard {...initialProps} />
        );
        expect(container).toMatchSnapshot();
    });

})
