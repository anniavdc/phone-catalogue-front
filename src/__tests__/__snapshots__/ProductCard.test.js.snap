// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard render 1`] = `
<div>
  <div
    class="product-card"
  >
    <p
      class="card-info"
    >
      59
    </p>
    <div
      class="card-image"
      style="background-image: url(https://image.png);"
    />
    <div>
      <h3
        class="card-title"
      >
        Galaxy 7
      </h3>
      <p
        class="card-content"
      >
        Samsung
      </p>
    </div>
  </div>
</div>
`;
